container_registry/container_analysis/src/sample/samples.go

```diff
@@ -0,0 +1,247 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// Samples for the Container Analysis golang libnraries: https://cloud.google.com/container-registry/docs/container-analysis
+package main
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	containeranalysis "cloud.google.com/go/devtools/containeranalysis/apiv1beta1"
+	pubsub "cloud.google.com/go/pubsub"
+	"golang.org/x/net/context"
+	"google.golang.org/api/iterator"
+	grafeaspb "google.golang.org/genproto/googleapis/devtools/containeranalysis/v1beta1/grafeas"
+	"google.golang.org/genproto/googleapis/devtools/containeranalysis/v1beta1/vulnerability"
+)
+
+// [START create_note]
+
+//createNote creates and returns a new vulnerability Note.
+func createNote(ctx context.Context, client *containeranalysis.GrafeasV1Beta1Client, noteID, projectID string) (*grafeaspb.Note, error) {
+	parent := "projects/" + projectID
+
+	vul := vulnerability.Vulnerability{}
+	// Details about the your vulnerability can be added here.
+	// For Example: vul.Severity = vulnerability.Severity_CRITICAL
+	noteVul := grafeaspb.Note_Vulnerability{Vulnerability: &vul}
+	note := grafeaspb.Note{Type: &noteVul}
+	req := &grafeaspb.CreateNoteRequest{Parent: parent, NoteId: noteID, Note: &note}
+	return client.CreateNote(ctx, req)
+}
+
+// [END create_note]
+
+// [START create_occurrence]
+
+//createsOccurrence creates and returns a new Occurrence of a previously created vulnerability Note.
+func createOccurrence(ctx context.Context, client *containeranalysis.GrafeasV1Beta1Client, imageUrl, parentNoteID, projectID string) (*grafeaspb.Occurrence, error) {
```

Comment:
You should just take in the noteName here, since the note's project isn't necessarily the same as the project you're creating the occurrence in. Check this for Java examples as well.

Reply:
good point. Should we change to using noteName in the other methods as well then too, to be consistent? (getNote and updateNote, etc)

Reply:
Hum maybe it's better to keep them separate because eventually we want to show people how to use the util that formats these (when I fix the generation to generate them). So maybe add noteProjectID as another param instead.